src/app/pages/ParaTimesPage/TransactionAmount/__tests__/index.test.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react'
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { paraTimesActions } from 'app/state/paratimes'
+import { useParaTimes, ParaTimesHook } from '../../useParaTimes'
+import { useParaTimesNavigation, ParaTimesNavigationHook } from '../../useParaTimesNavigation'
+import { TransactionAmount } from '..'
+import { ParaTime } from 'config'
+
+const mockDispatch = jest.fn()
+jest.mock('react-redux', () => ({
+  useSelector: jest.fn(),
+  useDispatch: () => mockDispatch,
+}))
+jest.unmock('react-i18next')
+jest.mock('../../useParaTimes')
+jest.mock('../../useParaTimesNavigation')
+
+describe('<TransactionAmount />', () => {
+  const mockUseParaTimesResult = {
+    balance: '1000000000',
+    balanceInBaseUnit: true,
+    isDepositing: true,
+    isEvmcParaTime: false,
+    isWalletEmpty: false,
+    paraTimeName: 'Cipher',
+    ticker: 'ROSE',
+    transactionForm: {
+      amount: '',
+      recipient: 'dummyAddress',
+      paraTime: ParaTime.Cipher,
+      privateKey: 'privateKey',
+    },
+    usesOasisAddress: true,
+  } as ParaTimesHook
+  const mockUseParaTimesNavigationResult = {} as ParaTimesNavigationHook
+
+  beforeEach(() => {
+    jest.mocked(useParaTimes).mockReturnValue(mockUseParaTimesResult)
+    jest.mocked(useParaTimesNavigation).mockReturnValue(mockUseParaTimesNavigationResult)
+  })
+
+  it('should render component', () => {
+    const { container } = render(<TransactionAmount />)
+
+    expect(container).toMatchSnapshot()
+  })
+
+  it('should render EVMc withdraw variant component', () => {
+    jest.mocked(useParaTimes).mockReturnValue({
+      ...mockUseParaTimesResult,
+      isDepositing: false,
+      isEvmcParaTime: true,
+      paraTimeName: 'Emerald',
+    } as ParaTimesHook)
+    render(<TransactionAmount />)
+
+    expect(screen.getByTestId('paraTime-content-description')).toMatchSnapshot()
+  })
+
+  it('should render disabled variant', () => {
+    jest.mocked(useParaTimes).mockReturnValue({
+      ...mockUseParaTimesResult,
+      balance: '0',
+      isWalletEmpty: true,
+    } as ParaTimesHook)
+    render(<TransactionAmount />)
+
+    expect(screen.getByText('The wallet is empty. There is nothing to withdraw.')).toBeInTheDocument()
+    expect(screen.getByPlaceholderText('0')).toBeDisabled()
+    expect(screen.getByRole('button', { name: 'MAX' })).toBeDisabled()
+    expect(screen.getByRole('button', { name: 'Next' })).toBeDisabled()
+  })
+
+  it('should set input value to max', () => {
+    const setTransactionForm = jest.fn()
+    jest.mocked(useParaTimes).mockReturnValue({ ...mockUseParaTimesResult, setTransactionForm })
+    render(<TransactionAmount />)
+
+    userEvent.click(screen.getByRole('button', { name: 'MAX' }))
+    expect(setTransactionForm).toHaveBeenCalledWith(expect.objectContaining({ amount: '1' }))
```

Comment:
Can you add precision tests with 1563114365108133939632n

Reply:
2493300888889 isn't enough :D js numbers have that precision

Reply:
updated in the second PR